tests/registration/test_registration.py

```diff
@@ -13,35 +15,136 @@
-            assert_that(response_registration.status_code, is_(201))
+            registration_response = AuthenticateAPI().registration(body=self.user_to_register)
+
+        with step("Checking the response code"):
+            assert_that(registration_response.status_code, is_(201))
+
+        with step("Checking the response body"):
+            response_json = registration_response.json()
+            assert_that(response_json["token"], is_not(empty()))
 
         with step("Getting info about the user in DB"):
             user_data = postgres.get_data_by_filter(
-                table="user_details", field="email", value=email
+                table="user_details", field="email", value=self.email
             )
 
         with step("Checking mapping data from the API request to the database"):
-            check_mapping_api_to_db(api_request=body, database_data=user_data[0])
+            check_mapping_api_to_db(api_request=self.user_to_register, database_data=user_data[0])
+
+    @pytest.mark.skip(reason="Bug in the API => wrong error message for not unique email")
+    @severity(severity_level="MAJOR")
+    @title("User registration with not unique email")
+    @description(
+        "WHEN the user submits data with a non-unique email for registration, "
+        "THEN the user receives an error message, and the user's information is not stored in the database"
+    )
+    def test_of_registration_email_uniqueness(self, postgres):
+        """
+        Test of registration of user with not unique email
+        Args:
+            postgres: fixture for working with the database
+        """
+        with step("Registration of user"):
+            registration_response = AuthenticateAPI().registration(body=self.user_to_register)
+            assert_that(registration_response.status_code, is_(201))
+
+        with step("Registration of user's duplicate"):
+            duplicate_registration_response = AuthenticateAPI().registration(body=self.user_to_register)
+
+        with step("Checking the response code"):
+            assert_that(duplicate_registration_response.status_code, is_(400))
+
+        with step("Checking the response body"):
+            expected_message = 'Email must be unique'
+            response_json = duplicate_registration_response.json()
+            assert_that(response_json["message"][0], is_(expected_message))
+
+        with step("Getting info about the user with not unique email in DB"):
+            user_data = postgres.get_data_by_filter(
+                table="user_details", field="email", value=self.email
+            )
+
+        with step("Checking that new user with duplicate email has not been registered "):
+            assert_that(user_data, has_length(1))
+            check_mapping_api_to_db(api_request=self.user_to_register, database_data=user_data[0])
+
+    fields = [
+        # TODO: check the error message for the email field
```

Comment:
let's clarify first, and then describe how it should be and we can add `pytest.mark.xfail`

Reply:
Commented for now, waiting for response.

Reply:
@yuliahar if you take responsibility that you won't forget and you will definitely fix it in the near future, then let's merge :)

Reply:
agreed :)